### PR TITLE
Update the Benchmarking docs and add workflow for making optimizations within S.P.CoreLib

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @layomia, @GrabYourPitchforks

### CODE REVIEW

File: Documentation/project-docs/benchmarking.md [hunk 1]
```diff
@@ -82,10 +82,65 @@ You should notice that given `.dll` file have been updated in the `CoreRun` fold
...
+
+        dotnet run -c Release -f netcoreapp3.0 -- -f * --coreRun "C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe" --artifacts ".\before"
+
+7. Go to the corresponding CoreCLR source folder where the API you want to change exists (for example `coreclr\src\System.Private.CoreLib\shared\System`)
+8. Apply the optimization that you want to test
+9. Rebuild System.Private.Corelib with your change (optionally adding `-skipnative` if the change is isolated to managed code):
+
+        C:\Projects\coreclr>build.cmd -release -skiptests -skipnative
+
+10. For the next step you have one of two options depending:
```
Comment (stephentoub) [inline Documentation/project-docs/benchmarking.md]: "depending"?  Depending on what?

File: Documentation/project-docs/benchmarking.md [hunk 2]
```diff
@@ -82,10 +82,65 @@ You should notice that given `.dll` file have been updated in the `CoreRun` fold
...
+  - OR manually copy over the relevant files from within the root of the coreclr output folder to where `CoreRun.exe` lives within corefx (excluding the subdirectories). This ends up being much faster than the first option and if the only thing that changed is   `System.Private.Corelib.dll`, just copy that over:
+
+          Copy from: C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release
+          To: C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\
+
+11. Run the benchmarks using `--coreRun` from the first step. Save the results in a dedicated folder.
+
+        dotnet run -c Release -f netcoreapp3.0 -- -f * --coreRun "C:\Projects\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\9.9.9\CoreRun.exe" --artifacts ".\after"
+
+12. Compare the results and repeat steps `8 - 11` until you are happy about the results.
```
Comment (stephentoub) [inline Documentation/project-docs/benchmarking.md]: There's a tool for comparing the results, right?

**Comment by stephentoub** [comment]
cc: @adamsitnik

**Comment by ahsonkhan** [comment]
Not waiting for CI since its just an md file change.

**Comment by adamsitnik** [comment]
@ahsonkhan big thanks for updating the docs!

File: Documentation/project-docs/benchmarking.md (hunk 2 above)
Comment (ahsonkhan) [reply Documentation/project-docs/benchmarking.md]: cc @adamsitnik, do we have such a tool?

**Pull request merged**